core/src/main/java/com/linecorp/armeria/server/docs/DocService.java

```diff
@@ -157,6 +167,24 @@ private static <T> Map<String, ListMultimap<String, T>> immutableCopyOf(
                 Entry::getKey, e -> ImmutableListMultimap.copyOf(e.getValue())));
     }
 
+    private static NamedTypeInfoProvider composeNamedTypeInfoProvider(
+            @Nullable NamedTypeInfoProvider namedTypeInfoProvider) {
+        return typeDescriptor -> {
+            if (namedTypeInfoProvider != null) {
+                // Respect user-defined provider first.
+                return namedTypeInfoProvider.newNamedTypeInfo(typeDescriptor);
```

Comment:
Sorry if I missed this discussion 😅 
Do we want to discard SPI loaded implementations if a user-defined provider has been specified? (since we're just returning instead of `orElse`).

Reply:
Oops. It is a bug. 😱